MinecraftClient/Protocol/ProtocolHandler.cs

```diff
@@ -483,21 +483,29 @@ private static LoginResult MicrosoftMCCLogin(string email, string password, out
         public static LoginResult MicrosoftBrowserLogin(out SessionToken session)
         {
             var ms = new XboxLive();
-            string[] askOpenLike =
+            string[] askOpenLink =
             {
                 "Copy the following link to your browser and login to your Microsoft Account",
                 ">>>>>>>>>>>>>>>>>>>>>>",
                 "",
                 ms.SignInUrl,
                 "",
                 "<<<<<<<<<<<<<<<<<<<<<<",
-                "When you see a blank page after signing in, copy the link from browser and paste it below"
+                "NOTICE: Once successfully logged in, you will see a blank page in your web browser.",
+                "Copy the contents of your browser's address bar and paste it below to complete the login process.",
             };
-            ConsoleIO.WriteLine(string.Join("\n", askOpenLike));
+            ConsoleIO.WriteLine(string.Join("\n", askOpenLink));
             string[] parts = { };
             while (true)
             {
                 string link = ConsoleIO.ReadLine();
+                if (string.IsNullOrEmpty(link))
+                {
+                    ConsoleIO.WriteLine("Login cancelled.");
+                    Program.Exit();

```

Comment:
Seems a bit harsh to hard quit here, better return a login status like cancelled or failed 😄

Reply:
Done.